packages/datagrid/src/components/RichCellEditor/RichCellEditor.component.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+import { SkeletonParagraph } from '@talend/design-system';
+import classNames from 'classnames';
+
+import { AgGridCellValue } from '../../types';
+
+import CellEditorDatalist from './components/CellEditorDatalist.component';
+import CellEditorTextarea from './components/CellEditorTextarea.component';
+
+import theme from './RichCellEditor.scss';
+
+type CellValue = string;
+
+interface RichCellEditorPropTypes {
+	initialValue: CellValue;
+	isLoading?: boolean;
+	hasSuggestions?: boolean;
+	onChange: (value: CellValue) => void;
+	onFilter?: (search: string) => Promise<AgGridCellValue[]>;
+	onCancel: () => void;
+}
+
+function RichCellEditor(props: RichCellEditorPropTypes) {
+	const { onChange, onFilter, onCancel, initialValue, hasSuggestions, isLoading } = props;
+	const [value, setValue] = useState(initialValue);
+
+	const handleInputChange = (
+		_: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+		newValue: CellValue,
+	): void => {
```

Comment:
```suggestion
	) => {
```
Not sure we need to set the return type, it should be inferred?

Reply:
I'm a bot